akka-http-cors/src/main/scala/ch/megard/akka/http/cors/scaladsl/settings/CorsSettings.scala

```diff
@@ -119,6 +119,28 @@ abstract class CorsSettings extends javadsl.settings.CorsSettings {
 
 object CorsSettings {
 
+  val prefix = "akka.http.cors"
+
+  def apply(config: Config) = fromSubConfig(config.getConfig(prefix))
+
+  def fromSubConfig(config: Config) = CorsSettings.Default(
+    allowGenericHttpRequests = config.getBoolean("allow-generic-http-requests"),
+    allowCredentials = config.getBoolean("allow-credentials"),
+    allowedOrigins = config.getStringList("allowed-origins").asScala match {
```

Comment:
Using pattern matching on List make this more readable.
```
allowedOrigins = config.getStringList("allowed-origins").asScala.toList match {
  case List("*") => HttpOriginRange.*
  case origins => HttpOriginRange.Default(origins.map(HttpOrigin(_)))
}
```

Same for `allowedHeaders`.